src/Core/Content/Flow/Dispatching/Storer/CustomerStorer.php

```diff
@@ -70,21 +72,13 @@ private function lazyLoad(StorableFlow $storableFlow): ?CustomerEntity
             return null;
         }
 
-        $criteria = new Criteria([$id]);
+        $criteria = $this->customerCriteriaBuilder->getCriteria($id);
 
         return $this->loadCustomer($criteria, $storableFlow->getContext(), $id);
     }
 
     private function loadCustomer(Criteria $criteria, Context $context, string $id): ?CustomerEntity
     {
-        $criteria->addAssociation('salutation');
-        $criteria->addAssociation('defaultBillingAddress.country');
-        $criteria->addAssociation('defaultBillingAddress.countryState');
-        $criteria->addAssociation('defaultBillingAddress.salutation');
-        $criteria->addAssociation('defaultShippingAddress.country');
-        $criteria->addAssociation('defaultShippingAddress.countryState');
-        $criteria->addAssociation('defaultShippingAddress.salutation');
-
         $event = new BeforeLoadStorableFlowDataEvent(
```

Comment:
I don't think going with two events (this one + your added one in the `criteriaBuilder`) is a good idea, because:
- the returned data / criteria should stay consistent, no matter if the trigger is from a flow or a normal HTTP endpoint. Having two different events can lead to inconsistencies
- It should stay simple, you can't expect from third parties to make their criteria adjustments in both events

So I would suggest the following:
- Move this event + dispatch inside the `criteriaBuilder`, which itself shouldn't be a break
- Deprecate it to either:
  - Rename it with 6.8
  - Replace it with your event for 6.8 if the arguments should also change